### ISSUE TITLE
POI names of some places double on map (but not all)

### ISSUE THREAD

**Issue by Maartensworld** [osmandapp/OsmAnd#21034]
### Description

Several names of my places are printed double on the map. Slightly above each other, making it impossible to read.  I checked, they are not double in my places and if I move the poi the double name moves also.
![Screenshot_20241014_151323_OsmAnd+](https://github.com/user-attachments/assets/07096977-f4d7-42be-94ec-bbd37b67b07f)

### Steps to reproduce

If I would know that, I could av9id it from happening

### Actual result

The name of the place (my places) is printed double on the map. But not all places have this.

### Expected result

The name should be printed only once

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:
Android/iOS version:
Device model:
Crash-Logs: ?
```
On a galaxy a9+ tablet

**Comment by yuriiurshuliak** [comment]
Could you please provide the coordinates of the locations where the Points of Interest (POIs) are duplicated? This will help us investigate the issue further.

**Comment by Maartensworld** [comment]
You misunderstand. It is not the POI themselves that are double. The name of the POI I make myself (my places) is printed double, half over each other (see the screen shot).
Those places are all over Europe.

But, as I said, not all of them and not all the time.

**Comment by yuriiurshuliak** [comment]
On my end, I haven’t encountered any duplicated POIs on the map. Unfortunately, based on your description, I’m not entirely sure how to reproduce this bug.

Could you clarify whether you are referring to Favorites or POIs? Is it possible that your Favorites are overlapping with POIs, causing the names to appear duplicated? 

Please provide the exact steps you follow to reproduce the issue. If possible, attach a video showing the process and how you interact with the object. Also, let me know which version of the app you are using.

**Comment by Maartensworld** [comment]
Hi.

Sorry, I meant "favorites" when I said "my places".

So, I attached a video that shows an example.

In de video you can see that the name is printed double (and becomes
unreadable).
If I then "switch off" the display of favorites (in "configure map", one of
the names disapears.
If I then switch off "point labels" the remaining name disapears.
If I then switch on "favorites" again, only the icon apears (without a
name) and if I switch on "point labels" again, the name gets printed double
again.

So, it seems the print comes from two separate places ("favorites" and
"point labels") but if "point labels" is not switched on, "favorites" will
not show its label.

Solve it by disconnecting labels for favorites from point labels or make
sure favorites does never print a label.

(Ar least, that is what it looks like).

BUT
It is not always the case and not always with the same favorites, so there
must be a third factor in play.

Now, I work a lot with tracks and routes, guiding them with waypoints
(which are favorites) so maybe that is the third factor.
But I have not been able to reproduce this problem. It just happens... or
not.

Now, I am trying to see which version I have, but can not find it ☺️.
It it the last version of OsmAnd+
OK, found it in the android settings (4.8.6)

I hope this helps.

**Comment by Maartensworld** [comment]
Third bit of info.
To find the version number I restarted osmand and since the screen was
still in the same location, i could see the favorite from the video I send
you, and now it is no longer double printed.

Tried toggling the settings I mentioned before, but it stays single printed.

To add to the mystery

**Comment by Maartensworld** [comment]
Hello,
I have tried to reply to the emails I received from yuriiurshuliak but they are blocked (response from remote server: 552 5.3.4 message size exceeds fixed limit.

Since that conversation is not listed here, I do not know how to reply.

**Comment by Maartensworld** [comment]
Just so you know: Your server send me a message that all my emails (that are now listed here) were blocked.
You might want to look into that.

**Comment by yuriiurshuliak** [comment]
The bug can be reproduced. To reproduce it, try creating a Favorite over an existing POI, then enable the "Point Labels (POI, Favorites)" option in Configure Map.

As a result, you’ll notice that the names of the POIs and Favorites are duplicated.

OsmAnd~ 4.9.0#3436m, released: 2024-10-14

https://github.com/user-attachments/assets/47aa4e87-ada5-4459-a3de-f1fc58f7ecac

**Comment by Maartensworld** [comment]
I understand that.  But that is not what is happening.
I don't create favorites over POI as that makes the map messy.

This happens with favorites created anywhere on the map.

**Comment by Maartensworld** [comment]
I have tried several times to add a video, but this system continuously blocks it.

**Comment by vshcherb** [comment]
It's not reproduced. Double names from map & poi - are known issue separate issue. Here doubling names comes from just Favorites?

**Comment by yuriiurshuliak** [comment]
@Maartensworld Please note that GitHub has a 10MB limit for video uploads. You can upload your video to a cloud storage service and share the link here. Alternatively, you can try compressing the video into a zip file or send it directly to our support email at support@osmand.net.

**Comment by Maartensworld** [comment]
Hi.
Yes, it still happens every now and then.
Just now I made the attached photo of a favorite I made.
As you can see, the POI name is there as it should and the favorite name is
doubly printed.
Once with a white backdrop and once with a transparent backdrop.

Strangly, only this favorite displays double at this time.  Others are fine.
But tomorrow this favorite might display fine and an other display double.

I can not find any logic in it.

Maarten

**Comment by yuriiurshuliak** [comment]
I wasn’t able to reproduce this behavior on either the nightly or release versions. I located the area from your screenshot, created a favorite there, and enabled "Favorites" and "Point Labels (POI, Favorites)" in the Configure map.

As a result, only a single name is displayed for favorites, with no duplication.

This suggests another factor may be influencing the issue, but we lack enough details to reproduce it as you experienced. You mentioned sending a video but haven’t provided it yet.  So could you please provide us with a recording showing how the bug occurs, along with an export of your profile settings where the issue appears and an export of any favorites displaying duplicate names?

You can upload the video to a cloud storage service and share the link here. Alternatively, compressing it into a zip file or emailing it to us directly at support@osmand.net also works.

To export your profile and favorites with the necessary data, go to Menu → Settings → Export.

OsmAnd~ 4.9.0#3591m, released: 2024-10-29

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/e3a6b59a-48ce-4171-a0ad-afa3396031f6"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by tmusolf** [comment]
I have recently been having the same issue with many of the favorites I create.  This has nothing to do with POIs.  It is not all favorites that I create, just some.  I have not been able to isolate steps to reproduce.


In the attached screen shot I created these two favorites with a couple of minures of each other. I moved their position slightly and edited their name after creating.  One has double text and the other does not.

OsmAnd+ 4.8.6, released: 2024-08-08

![Screenshot_20241101_171532_OsmAnd+.jpg](https://github.com/user-attachments/assets/ff6bc985-a866-441f-9a46-60e921352beb)

**Comment by vshcherb** [comment]
May be it's doubling Favorites and GPX and POI?   Let's check which layers are enabled

**Comment by tmusolf** [comment]
Is that something you are suggesting I check?  If so, where can I get you the needed info?

The map is displaying gpx files with waypts and tracks, but the points with double text were added as favorites in a named favorite folder.

With both favorites and point labels on you see the favorite icons and double text.

![Screenshot_20241106_070458_OsmAnd+.jpg](https://github.com/user-attachments/assets/9db79c4e-f717-4abb-9071-6c90c75d8d96)

Now turn off favorites switch.  Favorites icon and text with thick white border goes away, leaving just the thin border text.

![Screenshot_20241106_070614_OsmAnd+.jpg](https://github.com/user-attachments/assets/9f11b2c5-04d1-43e6-9805-b6444fda9452)

Turn on favorites, but turn off Point Labels you get the favorite icons, but no text of any kind.

![Screenshot_20241106_070822_OsmAnd+.jpg](https://github.com/user-attachments/assets/ce48472a-54bb-4fb1-a381-2ccd72d8091e)

**Comment by yuriiurshuliak** [comment]
I’ve been unable to reproduce the same behavior on my end. I tried creating multiple Favorites on the map with different map sources, overlays, underlay and GPX files with waypoints, as well as switching "Favorites" and "Point Labels (POI, Favorites)" on and off in the Configure map settings.

@tmusolf To help investigate further, could you please provide an export of your profile settings where the issue appears, along with an export of any Favorites that display duplicate names? You can compress it into a zip file or email it directly to support@osmand.net.

To export your profile and favorites, go to Menu → Settings → Export. Thank you!

OsmAnd~ 5.0.0#3674m, released: 2024-11-08

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/5a7e4af9-9bb7-407a-8254-f28953a654a3"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by tmusolf** [comment]
Since my last reply on this issue I have rebooted my phone and turned
groups of my places tracks and way points on and off.  I have also created
a new favorites folder and placed some favorites in that folder.

Checking the favorites that previously displayed the double text, they are
now displayed correctly.


I will zip up and send the config info you requested.

**Comment by vshcherb** [comment]
I think we will continue to investigate situation though it looks like it was a temporary glitch

**Comment by tmusolf** [comment]
This has recently happened again.  See attached two screen shots.  Yellow
waypt is the last added favorite. It was added then edited to correct the
spelling of it's name.  The red waypts are from a my places gpx file.

The yellow and one of the red waypts show double text, the other red waypt
does not.

I noticed the double text after editing the yellow favorite.

When dragging the map all text reverts to correct text with no double
text.  As long as your finger is held on the screen text remains correct.
Lift your finger and the text immediately returns to double text for the
yellow waypt and one red waypt.

None of the other favorites or gpx waypts are affected, just these two.

I have not restarted osmand nor rebooted the phone.

The first screen shot is without dragging and the second one is while my
finger is on the map and dragging.

**Comment by DmitryAlexei** [comment]
@tmusolf screenshots and videos cannot be attached to issues via email
please attach them using website

**Comment by tmusolf** [comment]
In reviewing the two screenshots I attached in the previous message they both still have double text.  The second image where I was dragging the map the double text was just "moved" to a much lower position.  Lifting your finger when the drag is complete causes the double text to move back to the previos, slightly offset position.
